src/common/datomish/db.cljc

```diff
@@ -832,9 +809,39 @@
 
 (defn <?q
   "Execute the provided query on the provided DB.
-   Returns a transduced pair-chan with one [[results] err] item."
+   Returns a transduced pair-chan with either:
+   * One [[results] err] item (for relation and collection find specs), or
+   * One [value err] item (for tuple and scalar find specs)."
   ([db find]
    (<?q db find {}))
   ([db find options]
-   (a/reduce (partial reduce-error-pair conj) [[] nil]
-             (<?run db find options))))
+   (let [unexpected (seq (clojure.set/difference (set (keys options)) #{:limit :order-by :inputs}))]
+    (when unexpected
+      (raise "Unexpected options: " unexpected {:bad-options unexpected})))
+   (let [{:keys [limit order-by inputs]} options
+         parsed (query/parse find)
+         context (-> db
+                   query-context
+                   (query/options-into-context limit order-by)
+                   (query/find-into-context parsed))
+
+         ;; We turn each row into either an array of values or an unadorned
+         ;; value. The row-pair-transducer does this work.
+         ;; The only thing to do to handle the full suite of find specs
+         ;; is to decide if we're then returning an array of transduced rows
+         ;; or just the first result.
+         row-pair-transducer (projection/row-pair-transducer context)
+         sql (query/context->sql-string context inputs)
+
+         first-only (context/scalar-or-tuple-query? context)
+         chan (chan (if first-only 1 50) row-pair-transducer)]
+
+     ;; Fill the channel.
+     (s/<?all-rows (.-sqlite-connection db) sql chan)
+
+     ;; If we only want the first result, great!
```

Comment:
`<?all-rows` puts `[result err]` pairs on to `chan`, so it would "look like" `([first-result nil])` in the `first-only` case, right?  That seems correct.